src/launch/linux/ioctl.rs

```diff
@@ -0,0 +1,63 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::impl_const_id;
+use crate::launch::types::*;
+use iocuddle::*;
+
+use std::marker::PhantomData;
+use std::os::unix::io::AsRawFd;
+
+// These enum ordinal values are defined in the Linux kernel
+// source code: include/uapi/linux/kvm.h
+impl_const_id! {
+    pub Id => u32;
+    Init = 0,
+}
+
+const ENC_OP: u64 = 0xc008aeba;
+
+// Note: the iocuddle::Ioctl::classic constructor has been used here because
+// KVM_MEMORY_ENCRYPT_OP ioctl was defined like this:
+//
+// _IOWR(KVMIO, 0xba, unsigned long)
+//
+// Instead of something like this:
+//
+// _IOWR(KVMIO, 0xba, struct kvm_sev_cmd)
+//
+// which would require extra work to wrap around the design decision for
+// that ioctl.
+
+/// Initialize the SEV platform context.
+pub const INIT: Ioctl<WriteRead, &Command<Init>> = unsafe { Ioctl::classic(ENC_OP) };
+
+#[repr(C)]
+pub struct Command<'a, T: Id> {
+    code: u32,
+    data: u64,
+    error: u32,
+    sev_fd: u32,
+    _phantom: PhantomData<&'a T>,
+}
+
+impl<'a, T: Id> Command<'a, T> {
+    pub fn from_mut(sev: &impl AsRawFd, subcmd: &'a mut T) -> Self {
```

Comment:
```suggestion
    pub fn from_mut(sev: &'a impl AsRawFd, subcmd: &'a mut T) -> Self {
```

I think we need to guarantee the lifetime of the file. Otherwise we could end up in a race condition where the file is closed, but we're still hanging on to the file descriptor (which could theoretically be reused).